pkg/core/resources/store/pagination_store.go

```diff
@@ -66,9 +66,19 @@ func (p *paginationStore) List(ctx context.Context, list model.ResourceList, opt
 		return err
 	}
 
-	for _, item := range fullList.GetItems() {
-		if opts.Filter(item) {
-			_ = filteredList.AddItem(item)
+	if len(opts.ResourceKeys) > 0 {
+		for _, rk := range opts.ResourceKeys {
+			for _, item := range fullList.GetItems() {
```

Comment:
I'm slightly concerned about O(n^2) here, but I guess we can leave it and see on the flamegraphs if this is visible or not.

Reply:
Right switch the loop would likely be more friendly to CPU caches. But doesn't it make sense for resourceKeys to be set and have this in the existing for loop?